Gruntfile.js

```diff
@@ -131,6 +131,16 @@ module.exports = function( grunt ) {
 			},
 		);
 
+		// If the script is executed within a GHA job in a PR, use the last PR commit hash instead of the one from the
+		// currently checked out merge commit.
+		if ( Boolean( process.env.LAST_PR_COMMIT_HASH ) ) {
+			spawnQueue[ 0 ] = {
+				cmd: 'echo',
+				// eslint-disable-next-line no-template-curly-in-string
+				args: [ `${ process.env.LAST_PR_COMMIT_HASH.slice( 0, 9 ) }` ],
+			};
+		}
+
```

Comment:
This works, but it seems a little bit of a roundabout way to do it it. It seems like if `process.env.LAST_PR_COMMIT_HASH` is set, then the first item could just be omitted from the `spawnQueue` from the first place, and then below it can do:

```js
const commitHash = process.env.LAST_PR_COMMIT_HASH ? process.env.LAST_PR_COMMIT_HASH.slice( 
0, 9 ) : stdout.shift()
```

This then skips having to run a subprocess just to echo back a string we already know.

Reply:
Yea I was thinking of something like that. You'd have to either `stdout.shift()` first to let `lsOutput` get the correct value, or set `lsOutput` to `stdout[2]` if `process.env.LAST_PR_COMMIT_HASH` is present.

Reply:
Yeah, so maybe:

```js
spawnQueue.push(
	{
		cmd: 'git',
		args: [ 'ls-files' ],
	},
);

if ( ! process.env.LAST_PR_COMMIT_HASH ) {
	spawnQueue.push(
		{
			cmd: 'git',
			args: [ '--no-pager', 'log', '-1', '--format=%h', '--date=short' ],
		}
	);
}
```

And then in `finalize`:

```js
const lsOutput = stdout.shift();
const commitHash = process.env.LAST_PR_COMMIT_HASH ? process.env.LAST_PR_COMMIT_HASH.slice( 
0, 9 ) : stdout.shift();
```

Reply:
Done in https://github.com/ampproject/amp-wp/pull/6259/commits/7e7c23d2d39606b2845df1f67fb25d829a653e55.